### ISSUE TITLE
ClassCastException

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#2779]
Found via google play console

```
java.lang.ClassCastException: 
  at com.owncloud.android.ui.activity.FileDisplayActivity.getShareFileFragment (FileDisplayActivity.java:1930)
  at com.owncloud.android.ui.activity.FileDisplayActivity.onCreateShareViaLinkOperationFinish (FileDisplayActivity.java:1824)
  at com.owncloud.android.ui.activity.FileDisplayActivity.onRemoteOperationFinish (FileDisplayActivity.java:1720)
  at com.owncloud.android.services.OperationsService$1.run (OperationsService.java:726)
``` 

@AndyScherzinger maybe you have an idea?
It seems that those are the steps to reproduce:
- enter file details view
- share a file via link
- (somehow) navigate away from detail view (share via link is a blocking dialog)
- crash as second fragment is no longer a FileDetailFragment

We can easily have a check prior casting, but still it is strange how this can be triggered at all.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky This might happen via the bottom sheet menu I changed and fixed via https://github.com/nextcloud/android/commit/0745348ce0a6da6534f4b6968d6a35ef2168f577 (so fixed on master) so I can't say if this happens on a production version but it did during development (which might still get reported to the dev console at Google!). So do we know the app-version?

And yes, the only way is to check for the fragment to be available since with the change I am refering to this had to be introduced anyways since the link creation can now be triggered through other fragments thus we do check if the fragment is present and if so refresh its data.

So the only question would be, if the null-check should be backported for 3.2.1 or if 3.3.0 is fine (since it is fixed on master already)

**Comment by tobiasKaminsky** [comment]
It was 3.2.0.
If it is a small and easy fix, I would like to have it in 3.2.1 as a known crash should not kept without real intention until 3.3.0.

**Comment by AndyScherzinger** [comment]
It is a small and easy fix so I'd also vote for having it in 3.2.1 :+1:
